zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java

```diff
@@ -281,9 +275,14 @@ public InterpreterSetting(InterpreterSetting o) {
     this.conf = o.getConf();
   }
 
-  private void createLauncher() {
+  private void createLauncher(Properties properties) {
     if (group.equals("spark")) {
-      this.launcher = new SparkInterpreterLauncher(this.conf);
+      if (properties.getProperty("master") != null
+              && properties.getProperty("master").startsWith("k8s")) {
```

Comment:
I think the proper match should be `k8s://`
https://github.com/apache-spark-on-k8s/spark/pull/498/files#diff-817de38fee3505bdca2e40ce393857f6R123